client/lib/feed-store/index.js

```diff
@@ -89,6 +94,17 @@ function receiveFeeds( newFeeds ) {
 	} ) );
 }
 
+function discoverFeeds( data, feedUrl ) {
+	if ( typeof data !== 'undefined' ) {
+		data.feeds.map( function( feed ) {
+			if ( feed.feed_ID ) {
+				feedsByUrl[ feedUrl ] = feed.feed_ID;
```

Comment:
ah, you're really reducing here, not mapping. how could you implement this in terms of `reduce`?